src/sentry/api/endpoints/client_state.py

```diff
@@ -0,0 +1,55 @@
+# This endpoint helps managing persisted client state with a TTL for a member, organization or user
+
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint
+from sentry.api.exceptions import ParameterValidationError
+from sentry.utils import json, redis
+
+STATE_CATEGORIES = {
+    "onboarding": {
+        "ttl": 30 * 24 * 60 * 60,  # the time in seconds that the state will be persisted
+        "scope": "member",  # Can be "user", "org", or "user-org"
+    }
+}
+
+
+class ClientStateEndpoint(Endpoint):
+    permission_classes = (IsAuthenticated,)
+
+    def get_key(self, request: Request, category):
+        scope = STATE_CATEGORIES[category]["scope"]
+        org = request.query_params.get("org")
```

Comment:
@Neo-Zhixing We definitely need validation that the user has at least `org:read` access or else this is a security vulnerability. Honestly, it might make sense to just make the organization slug part of the route so you can use the `OrganizationEndpoint` class which will automatically check organization access. I don't honestly see an obvious use case for the cache where just the user is the key and even if do later, we can make another endpoint that doesn't depend on the organization to access the cache.